binary-decode.go

```diff
@@ -115,11 +115,12 @@ func (cdc *Codec) decodeReflectBinary(bz []byte, info *TypeInfo, rv reflect.Valu
 			}
 			rv.SetInt(num)
 		} else {
-			num, _n, err = DecodeVarint(bz)
+			var unum uint64
+			unum, _n, err = DecodeUvarint(bz)
 			if slide(&bz, &n, _n) && err != nil {
 				return
 			}
-			rv.SetInt(num)
+			rv.SetInt(int64(unum))
```

Comment:
should we also check that `unum <= math.MaxInt64` ?

Reply:
Thanks. Yes, you are right, we should!